pygen/pygen_src/riscv_asm_program_gen.py

```diff
@@ -252,10 +258,53 @@ def gen_init_section(self, hart):
             string = pkg_ins.indent + "j main"
             self.instr_stream.append(string)
 
+    # Setup MISA based on supported extensions
     def setup_misa(self):
-        # TO DO
-        misa = 0b01000000
-        self.instr_stream.append("{}li x{}, {}".format(pkg_ins.indent, cfg.gpr[0].value, hex(misa)))
+        misa = BitArray(rcs.XLEN)
+        if rcs.XLEN == 32:
+            misa = BitArray(hex='0x40000000')
+        elif rcs.XLEN == 64:
+            misa = BitArray(hex='0x80000000')
+        else:
+            misa = BitArray(hex='0xc0000000')
```

Comment:
small style nit: maybe it would be a little cleaner to zero out `misa` when you declare it on line 263; then you could simply index into bits `[xlen-1:xlen-2]` and set those two bits directly based on the XLEN ?

Reply:
hi @udinator 
I have changed the BitArray to vsc type  `misa = vsc.bit_t(rcs.XLEN)` in order to use the slicing similar to the SV like syntax.
It helps me to use the slicing here as well ` misa[rcs.XLEN - misa_ext_t.MISA_EXT_C - 1] = 1`  to `misa[misa_ex_t.MISA_EXT_C]`.
I hope this makes sense to you.